src/s3/async_put_object.rs

```diff
@@ -120,12 +116,23 @@ impl<'a> AsyncWrite for AsyncPutObject<'a> {
                         }
                     }
                     Poll::Pending => {
-                        // Tell the async executor that we're ready to try again whenever it is!
-                        cx.waker().wake_by_ref();
```

Comment:
This call to `wake_by_ref` seems unnecessary because we already poll the underlying future in this code path, and it will schedule the wakeup for us. (Or at least, commenting out this line of code didn't seem to change the behaviour at all).